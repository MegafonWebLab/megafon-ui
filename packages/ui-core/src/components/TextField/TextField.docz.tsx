import * as React from 'react';

export const TextFieldWrapper = ({ children }) => (
    <div style={{ width: 250 }}>{children}</div>
);
