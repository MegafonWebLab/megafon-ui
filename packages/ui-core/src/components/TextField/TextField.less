@import '~styles/base.less';

@b: e('.mfui-text-field');

@{b} {
    .fontFamily();
    color: @soggyAsphalt;
    font-size: 15px;
    line-height: 20px;

    &__field {
        font-family: inherit;
        position: relative;
        z-index: 1;
        font-size: 15px;
        height: 48px;
        width: 100%;
        box-sizing: border-box;
        outline: none;
        cursor: pointer;
        padding: 13px 20px 13px 20px;
        line-height: 20px;
        background-color: transparent;
        opacity: 1;
        border-radius: 0;
        -webkit-appearance: none;
        -moz-appearance: none;
        box-shadow: none;
        overflow: auto;
        margin: 0;
    }
    &__placeholder {
        color: lighten(@soggyAsphalt, 25%);
        padding-left: 20px;
        font-weight: 400;
        user-select: none;
        position: absolute;
        left: 0;
        top: 50%;
        transform: translateY(-50%);
        z-index: 0;
    }
    input::-webkit-input-placeholder {
        color: lighten(@soggyAsphalt, 50%);
    }
    input::-moz-placeholder {
        color: lighten(@soggyAsphalt, 50%);
    }
    input:-moz-placeholder {
        color: lighten(@soggyAsphalt, 50%);
    }
    input:-ms-input-placeholder {
        color: lighten(@soggyAsphalt, 50%);
    }
    input::-ms-clear, input::-ms-reveal {
        display: none;
    }
    &__field_big-space {
        word-spacing: 4px;
    }
    &__field-wrapper {
        position: relative;
        flex: 1 0 100px;
        background: @clearWhite;
    }

    &__icon-box {
        box-sizing: border-box;
        position: absolute;
        z-index: 0;
        top: 0;
        right: 0;
        bottom: 0;
        display: flex;
        padding-right: 16px;
        justify-content: center;
        align-items: center;
        cursor: pointer;
        text-align: center;
    }
    &__icon-box_password {
        z-index: 3;
    }
    &__icon-box_custom {
        z-index: 3;
    }

    &__icon {
        width: 32px;
        height: 32px;
        display: block;
        user-select: none;
    }

    &__text {
        font-size: 12px;
        line-height: 16px;
        margin-top: 4px;
    }
}

//mod size:
@{b} {
    &_size_large & {
        &__field {
            @media @desktop {
                height: 60px;
                padding: 19px 25px 19px 25px;
            }
        }
        &__icon-box {
            @media @desktop {
                padding-right: 24px;
            }
        }
    }
}

//mod color:
@{b} {
    &_color_default & {
        &__field {
            color: @soggyAsphalt;
            border: 1px solid @spbSky2;
            background-color: transparent;
        }
        &__field-wrapper_no-touch:hover @{b}__field,
        &__field:focus {
            border-color: @soggyAsphalt;
        }
    }

    &_color_default&_valid & {
        &__field-wrapper_no-touch:hover @{b}__field,
        &__field {
            border-color: @green;
        }
        &__icon {
            fill: @green;
        }
    }

    &_color_default&_error & {
        &__field-wrapper:hover @{b}__field,
        &__field {
            border-color: @fury;
        }
        &__icon {
            fill: @fury;
        }
        &__text_error {
            color: @fury;
        }
    }

    &_color_white & {
        &__field {
            color: @soggyAsphalt;
            border: 1px solid @clearWhite;
        }
        &__field-wrapper:hover @{b}__field,
        &__field:focus {
            border-color: @soggyAsphalt;
        }
    }
    &_color_white&_error & {
        &__field-wrapper:hover @{b}__field,
        &__field {
            border-color: @fury;
        }
        &__icon {
            fill: @fury;
        }
        &__text_error {
            color: @fury;
        }
    }
    &__text_success {
        color: @green;
    }

}

//mod disabled
@{b} {
    &_disabled & {
        &__field-wrapper {
            background-color: @spbSky1 !important;
        }
        &__field {
            border-color: @spbSky1 !important;
            cursor: default !important;
            color: @soggyAsphalt !important;
            -webkit-text-fill-color: @soggyAsphalt !important;
        }
        &__icon-box {
            cursor: default !important;
        }
    }
}

//mod icon
@{b} {
    &_icon &__field {
        padding-right: 55px;
    }
    &_icon&_size_large &__field {
        @media @desktop {
            padding-right: 65px;
        }
    }
}

//mod password
@{b} {
    &_password &__field, &_password &__placeholder {
        font: small-caption;
        font-size: 19px;
    }
}
