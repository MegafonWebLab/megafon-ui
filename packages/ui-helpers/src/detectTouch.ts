export default function detectTouch(): boolean {
    if (typeof window === 'undefined') {
        return false;
    }

    const { DocumentTouch } = window;
    const prefixes: string[] = ' -webkit- -moz- -o- -ms- '.split(' ');
    const mq = (queryParam: string): boolean => window.matchMedia(queryParam).matches || false;

    if ('ontouchstart' in window || (DocumentTouch && document instanceof DocumentTouch)) {
        return true;
    }

    const query: string = ['(', prefixes.join('touch-enabled),('), 'heartz', ')'].join('');

    return mq(query);
}
