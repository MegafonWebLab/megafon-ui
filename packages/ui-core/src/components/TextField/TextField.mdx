---
name: TextField
menu: Components
---
import { Playground, PropsTable } from 'docz';
import TextField from './TextField';

export const wrapper = (width = 250) => ({ children }) => {
    return (
        <div style={{width: `${width}px`}}>
            {children}
        </div>
    )
}

# TextField

## Properties

<PropsTable of={TextField} />

### Classes property keys
field        | description
-------------|----------
*input*       |  Class that is passed to the input

## API

Method  | Description
------- | ---
blur    | Trigger blur event on input field.
focus   | Trigger focus event on input field.

**Use TextField's 'ref' prop for API methods call:**

```
class SearchForm extends Component {
    this.textField = undefined;

    handleSubmit() {
        textField.blur();
    }

    render() {
        return (
            <Form onSubmit={this.handleSubmit}>
                <TextField ref={(instance) => { this.textField = instance; }} />
            </Form>
        );
    }
}
```

## Basic usage

<Playground wrapper={wrapper()}>
    <TextField name="name"/>
</Playground>

## Mask usage

<Playground wrapper={wrapper()}>
    <TextField
        placeholder="+7 (999) 999-99-99"
        mask="+7 (999) 999-99-99"
        maskChar="_"
        label="Phone number"
        id="2"
        classes={{input: 'parent__inputClass'}}
    />
</Playground>

## Error usage

<Playground wrapper={wrapper()}>
    <TextField error={true} noticeText="Name is required" />
</Playground>

## Valid usage

<Playground wrapper={wrapper()}>
        <TextField valid={true} />
</Playground>

## Comment usage

<Playground wrapper={wrapper()}>
    <TextField commentText="Comment" />
</Playground>

## Success text usage

<Playground wrapper={wrapper()}>
    <TextField successText="Field is correct" />
</Playground>

## Max length text usage

<Playground wrapper={wrapper()}>
    <TextField maxLength={10} />
</Playground>

## Disable usage

<Playground wrapper={wrapper()}>
    <TextField disabled={true} />
</Playground>

## Password type usage

<Playground wrapper={wrapper()}>
    <TextField name="name" placeholder="password" type="password"/>
</Playground>

## Increased size of space between words usage
<Playground wrapper={wrapper()}>
    <TextField bigSpace={true} />
</Playground>
