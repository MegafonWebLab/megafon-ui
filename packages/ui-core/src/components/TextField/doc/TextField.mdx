---
name: TextField
menu: Компоненты
route: components/textfield
---

import { Props } from 'docz';
import TextField from '../TextField';
import Example from './TextField.example.mdx';

# TextField

<CopyToClipboardBox text="import { TextField } from '@megafon/ui-core';"/>

<DoczTabs>
    <Example />
    <Props of={TextField} />
</DoczTabs>
