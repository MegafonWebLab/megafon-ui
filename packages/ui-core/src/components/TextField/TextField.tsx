import * as React from 'react';
import * as PropTypes from 'prop-types';
import './TextField.less';
import InputMask from 'react-input-mask';
import cnCreate from 'utils/cn';
import * as equal from 'deep-equal';
import CheckedIcon from 'icons/System/24/Checked_24.svg';
import ErrorIcon from 'icons/System/24/Cancel_24.svg';
import Hide from 'icons/Basic/24/Hide_24.svg';
import Show from 'icons/Basic/24/Show_24.svg';
import detectTouch from 'utils/detectTouch';
import InputLabel from 'components/InputLabel/InputLabel';

interface ITextFieldProps {
    /** Field title */
    label?: string;
    /** Field color scheme */
    color?: 'default' | 'white';
    /** Field size */
    size?: 'large';
    /** Error/Notice text */
    noticeText?: string;
    /** Comment text */
    commentText?: string;
    /** Success text */
    successText?: string;
    /** Not show icon of state */
    isHideIcon?: boolean;
    /** Autofocus */
    autoFocus?: boolean;
    /** Autocomplete */
    autocomplete?: 'on' | 'off';
    /** Validation passed */
    valid?: boolean;
    /** Validation error */
    error?: boolean;
    /** Disable field */
    disabled?: boolean;
    /** Required field */
    required?: boolean;
    /** Type - property tag <input> */
    type?: 'text' | 'password' | 'tel' | 'email';
    /** Field name */
    name?: string;
    /** Placeholder */
    placeholder?: string;
    /** Html id attribute */
    id?: string;
    /** Field value */
    value?: string;
    /** Max length  */
    maxLength?: number;
    /** Default value */
    defaultValue?: string;
    /** Custom icon */
    customIcon?: JSX.Element;
    /** Mask */
    mask?: string;
    /** Split symbol for mask */
    maskChar?: string;
    /** Increased size of space between words in the text box */
    bigSpace?: boolean;
    /** Custom classname */
    className?: string;
    /** Change handler */
    onChange?(e: React.SyntheticEvent<EventTarget>): void;
    /** Blur handler */
    onBlur?(e: React.SyntheticEvent<EventTarget>): void;
    /** Focus handler */
    onFocus?(e: React.SyntheticEvent<EventTarget>): void;
    /** KeyUp handler */
    onKeyUp?(e: React.SyntheticEvent<EventTarget>): void;
}

interface ITextFieldState {
    isPasswordHidden: boolean;
}

const cn = cnCreate('mfui-text-field');
class TextField extends React.Component<ITextFieldProps, ITextFieldState> {
    static propTypes = {
        label: PropTypes.string,
        color: PropTypes.oneOf(['default', 'white']),
        size: PropTypes.oneOf(['large']),
        noticeText: PropTypes.string,
        commentText: PropTypes.string,
        successText: PropTypes.string,
        isHideIcon: PropTypes.bool,
        autoFocus: PropTypes.bool,
        autocomplete: PropTypes.oneOf(['on', 'off']),
        valid: PropTypes.bool,
        error: PropTypes.bool,
        disabled: PropTypes.bool,
        required: PropTypes.bool,
        type: PropTypes.oneOf(['text', 'password', 'tel', 'email']),
        name: PropTypes.string,
        placeholder: PropTypes.string,
        id: PropTypes.string,
        value: PropTypes.string,
        maxLength: PropTypes.number,
        defaultValue: PropTypes.string,
        customIcon: PropTypes.element,
        mask: PropTypes.string,
        maskChar: PropTypes.string,
        bigSpace: PropTypes.bool,
        className: PropTypes.string,
        onChange: PropTypes.func,
        onBlur: PropTypes.func,
        onFocus: PropTypes.func,
        onKeyUp: PropTypes.func,
    };

    static defaultProps: Partial<ITextFieldProps> = {
        autocomplete: 'off',
        type: 'text',
        color: 'default',
        isHideIcon: false,
    };

    inputNode: any;
    isTouch: boolean = detectTouch();

    constructor(props: ITextFieldProps) {
        super(props);

        this.state = {
            isPasswordHidden: true,
        };
    }

    shouldComponentUpdate(nextProps: ITextFieldProps, nextState: ITextFieldState) {
        return !(equal(this.props, nextProps) && equal(this.state, nextState));
    }

    blur = () => {
        return this.inputNode.blur();
    }

    focus = () => {
        return this.inputNode.focus();
    }

    handleEyeToggle = () => {
        this.setState(
            (prevState: ITextFieldState) => {
                return { isPasswordHidden: !prevState.isPasswordHidden };
            }
        );
    }

    renderValidIcon() {
        return (
            <div
                className={cn('icon-box')}
            >
                <CheckedIcon className={cn('icon')} />
            </div>
        );
    }

    renderErrorIcon() {
        return (
            <div
                className={cn('icon-box')}
            >
                <ErrorIcon className={cn('icon')} />
            </div>
        );
    }

    renderCustomIcon() {
        return (
            <div
                className={cn('icon-box', { custom: !!this.props.customIcon })}>
                {this.props.customIcon}
            </div>
        );
    }

    renderPasswordFieldIcon() {
        const { isPasswordHidden } = this.state;

        return (
            <div
                className={cn('icon-box', { 'password': true })}
                onClick={this.handleEyeToggle}
            >
                {isPasswordHidden
                    ? <Hide className={cn('icon')} />
                    : <Show className={cn('icon')} />
                }
            </div>
        );
    }

    addInputNode = (node: HTMLInputElement ) => this.inputNode = node;

    renderInputElem(isPasswordType: boolean) {
        const { isPasswordHidden } = this.state;
        const params = {
            disabled: this.props.disabled,
            name: this.props.name,
            id: this.props.id,
            placeholder: this.props.placeholder,
            onChange: this.props.onChange,
            onBlur: this.props.onBlur,
            onFocus: this.props.onFocus,
            onKeyUp: this.props.onKeyUp,
            autoFocus: this.props.autoFocus,
            defaultValue: this.props.defaultValue,
            maxLength: this.props.maxLength,
            value: this.props.value,
            type: isPasswordType && !isPasswordHidden ? 'text' : this.props.type,
            required: this.props.required,
            autoComplete: this.props.autocomplete,
            className: cn('field', {
                'big-space': this.props.bigSpace,
            }),
        };

        if (this.props.mask) {
            return (
                <InputMask
                    {...params}
                    inputRef={this.addInputNode}
                    mask={this.props.mask}
                    maskChar={this.props.maskChar}
                />
            );
        }

        return <input ref={this.addInputNode} {...params} />;
    }

    render() {
        const {
            isHideIcon, customIcon, error, color,
            valid, disabled, size, className,
            commentText, successText, noticeText, type,
            label, id,
        } = this.props;

        const isAnyIcon = !isHideIcon && (!!customIcon || error || valid);
        const isStatusIcon = !isHideIcon && !customIcon;
        const isPasswordType = type === 'password';

        return (
            <div
                className={cn('', {
                    valid: valid,
                    error: error,
                    icon: isAnyIcon,
                    password: isPasswordType,
                    disabled,
                    color,
                    size,
                }, className)}
            >
                {label && <InputLabel label={label} id={id} />}
                <div
                    className={cn('field-wrapper', { 'no-touch': !this.isTouch })}
                >
                    <div>{this.renderInputElem(isPasswordType)}</div>
                    {customIcon && this.renderCustomIcon()}
                    {isStatusIcon && valid && this.renderValidIcon()}
                    {isStatusIcon && error && this.renderErrorIcon()}
                    {isPasswordType && this.renderPasswordFieldIcon()}
                </div>
                {(error || valid) && noticeText &&
                    <div
                        className={cn('text', { error: true })}
                        dangerouslySetInnerHTML={{ __html: noticeText }}
                    />
                }
                {commentText &&
                    <div className={cn('text', { comment: true })}>{commentText}</div>}
                {successText &&
                    <div className={cn('text', { success: true })}>{successText}</div>}
            </div>
        );
    }
}

export default TextField;
