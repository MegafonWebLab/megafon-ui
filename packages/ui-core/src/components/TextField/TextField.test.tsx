import * as React from 'react';
import { shallow } from 'enzyme';
import TextField from './TextField';
import detectTouch from 'utils/detectTouch';
import { ITextFieldProps } from './TextField';

jest.mock('utils/detectTouch', () => jest.fn().mockReturnValue(false));

jest.mock('react-input-mask', () => {
    return {
        default() {},
    };
});

const inputClassName = '.mfui-text-field__field';

const props: ITextFieldProps = {
    label: <div>label</div>,
    color: 'default',
    size: 'large',
    noticeText: 'notice text',
    commentText: 'comment text',
    successText: 'success text',
    autoFocus: true,
    autocomplete: 'on',
    disabled: true,
    required: true,
    type: 'text',
    name: 'name',
    placeholder: 'placeholder',
    id: 'id',
    value: 'value',
    maxLength: 10,
    defaultValue: 'default value',
    customIcon: <div>custom icon</div>,
    bigSpace: true,
    className: 'class name',
    classes: {input: 'parent__inputClass'},
};

function simulateEvent(eventName: string, eventType: string, params: {}) {
    const data = 'data';
    const mock = jest.fn();
    const handler = { [eventName]: mock };
    const wrapper = shallow(<TextField {...handler} {...params} />);

    wrapper.find(inputClassName).simulate(eventType, data);

    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toBeCalledWith(data);
}

describe('<TextField />', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('renders without props', () => {
        const wrapper = shallow(<TextField />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders with props', () => {
        const wrapper = shallow(<TextField {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders with mask', () => {
        const wrapper = shallow(<TextField {...props} mask="(999) 999-99-99" maskChar=" " />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without no-touch mode', () => {
        const typedDetectTouch = detectTouch as jest.Mock<() => boolean>;

        typedDetectTouch.mockImplementation(() => true);

        const wrapper = shallow(<TextField />);

        expect(wrapper).toMatchSnapshot();
    });

    describe('icon mod', () => {
        it('renders with mod when customIcon passed', () => {
            const wrapper = shallow(
                <TextField isHideIcon={false} customIcon={<div>icon</div>} error={false} valid={false} />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('renders with mod when error is true', () => {
            const wrapper = shallow(
                <TextField isHideIcon={false} customIcon={undefined} error={true} valid={false} />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('renders with mod when valid is true', () => {
            const wrapper = shallow(
                <TextField isHideIcon={false} customIcon={undefined} error={false} valid={true} />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('renders without mod when isHideIcon is true', () => {
            const wrapper = shallow(<TextField isHideIcon={true} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('status icons', () => {
        it('renders when valid or error is true', () => {
            const wrapper = shallow(
                <TextField isHideIcon={false} customIcon={undefined} error={true} valid={true} />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('not renders when valid or error is false', () => {
            const wrapper = shallow(
                <TextField isHideIcon={false} customIcon={undefined} error={false} valid={false} />
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('password type', () => {
        it('renders with default input', () => {
            const wrapper = shallow(<TextField type="password" />);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders with mask input', () => {
            const wrapper = shallow(<TextField type="password" mask="(999) 999-99-99" maskChar=" " />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('calls handlers on different inputs', () => {
        const maskParams = { mask: '(999) 999-99-99', maskChar: ' ' };

        simulateEvent('onChange', 'change', {});
        simulateEvent('onChange', 'change', maskParams);

        simulateEvent('onBlur', 'blur', {});
        simulateEvent('onBlur', 'blur', maskParams);

        simulateEvent('onFocus', 'focus', {});
        simulateEvent('onFocus', 'focus', maskParams);

        simulateEvent('onKeyUp', 'keyup', {});
        simulateEvent('onKeyUp', 'keyup', maskParams);
    });

    it('calls handleEyeToggle', () => {
        const wrapper = shallow<TextField>(<TextField />);

        wrapper.setState({ isPasswordHidden: false });
        wrapper.instance().handleEyeToggle();

        expect(wrapper.state().isPasswordHidden).toBeTruthy();
    });

    it('calls onCustomIconClick handler', () => {
        const handler = jest.fn();
        const wrapper = shallow<TextField>(
            <TextField customIcon={<div>custom icon</div>} onCustomIconClick={handler} />
        );

        wrapper.find('.mfui-text-field__icon-box_custom').simulate('click');

        expect(handler).toHaveBeenCalledTimes(1);
    });
});
