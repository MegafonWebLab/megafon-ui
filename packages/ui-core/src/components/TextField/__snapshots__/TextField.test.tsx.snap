// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> icon mod renders with mod when customIcon passed 1`] = `
<div
  className="mfui-text-field mfui-text-field_icon mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
    <div
      className="mfui-text-field__icon-box mfui-text-field__icon-box_custom"
    >
      <div>
        icon
      </div>
    </div>
  </div>
</div>
`;

exports[`<TextField /> icon mod renders with mod when error is true 1`] = `
<div
  className="mfui-text-field mfui-text-field_error mfui-text-field_icon mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
    <div
      className="mfui-text-field__icon-box"
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> icon mod renders with mod when valid is true 1`] = `
<div
  className="mfui-text-field mfui-text-field_valid mfui-text-field_icon mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
    <div
      className="mfui-text-field__icon-box"
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> icon mod renders without mod when isHideIcon is true 1`] = `
<div
  className="mfui-text-field mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> password type renders with default input 1`] = `
<div
  className="mfui-text-field mfui-text-field_password mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="password"
      />
    </div>
    <div
      className="mfui-text-field__icon-box mfui-text-field__icon-box_password"
      onClick={[Function]}
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> password type renders with mask input 1`] = `
<div
  className="mfui-text-field mfui-text-field_password mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <[object Object]
        autoComplete="off"
        className="mfui-text-field__field"
        inputRef={[Function]}
        mask="(999) 999-99-99"
        maskChar=" "
        type="password"
      />
    </div>
    <div
      className="mfui-text-field__icon-box mfui-text-field__icon-box_password"
      onClick={[Function]}
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> renders with mask 1`] = `
<div
  className="mfui-text-field mfui-text-field_icon mfui-text-field_disabled mfui-text-field_color_default mfui-text-field_size_large class name"
>
  <InputLabel
    htmlFor="id"
  >
    <div>
      label
    </div>
  </InputLabel>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <[object Object]
        autoComplete="on"
        autoFocus={true}
        className="mfui-text-field__field mfui-text-field__field_big-space parent__inputClass"
        defaultValue="default value"
        disabled={true}
        id="id"
        inputRef={[Function]}
        mask="(999) 999-99-99"
        maskChar=" "
        maxLength={10}
        name="name"
        placeholder="placeholder"
        required={true}
        type="text"
        value="value"
      />
    </div>
    <div
      className="mfui-text-field__icon-box mfui-text-field__icon-box_custom"
    >
      <div>
        custom icon
      </div>
    </div>
  </div>
  <div
    className="mfui-text-field__text mfui-text-field__text_comment"
  >
    comment text
  </div>
  <div
    className="mfui-text-field__text mfui-text-field__text_success"
  >
    success text
  </div>
</div>
`;

exports[`<TextField /> renders with props 1`] = `
<div
  className="mfui-text-field mfui-text-field_icon mfui-text-field_disabled mfui-text-field_color_default mfui-text-field_size_large class name"
>
  <InputLabel
    htmlFor="id"
  >
    <div>
      label
    </div>
  </InputLabel>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="on"
        autoFocus={true}
        className="mfui-text-field__field mfui-text-field__field_big-space parent__inputClass"
        defaultValue="default value"
        disabled={true}
        id="id"
        maxLength={10}
        name="name"
        placeholder="placeholder"
        required={true}
        type="text"
        value="value"
      />
    </div>
    <div
      className="mfui-text-field__icon-box mfui-text-field__icon-box_custom"
    >
      <div>
        custom icon
      </div>
    </div>
  </div>
  <div
    className="mfui-text-field__text mfui-text-field__text_comment"
  >
    comment text
  </div>
  <div
    className="mfui-text-field__text mfui-text-field__text_success"
  >
    success text
  </div>
</div>
`;

exports[`<TextField /> renders without no-touch mode 1`] = `
<div
  className="mfui-text-field mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> renders without props 1`] = `
<div
  className="mfui-text-field mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> status icons not renders when valid or error is false 1`] = `
<div
  className="mfui-text-field mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField /> status icons renders when valid or error is true 1`] = `
<div
  className="mfui-text-field mfui-text-field_valid mfui-text-field_error mfui-text-field_icon mfui-text-field_color_default"
>
  <div
    className="mfui-text-field__field-wrapper mfui-text-field__field-wrapper_no-touch"
  >
    <div>
      <input
        autoComplete="off"
        className="mfui-text-field__field"
        type="text"
      />
    </div>
    <div
      className="mfui-text-field__icon-box"
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
    <div
      className="mfui-text-field__icon-box"
    >
      <test-file-stub
        className="mfui-text-field__icon"
      />
    </div>
  </div>
</div>
`;
